sites/game-client/map-renderer.mjs

```diff
@@ -139,6 +139,44 @@ class MapRenderer {
     };
   }
 
+  /**
+   * Determine whether the target is inside the viewport.
+   * @param {string} mapName - The map the client using.
+   * @param {object} canvasCoord1 - The target's left-top coordinate (x, y) relative to canvas.
+   * @param {object} canvasCoord2 - The target's right-bottom coordinate (x, y) relative to canvas.
+   * @return {bool} isInside - return true if the target is inside the viewport.
+   */
+  insideViewport(mapName, canvasCoord1, canvasCoord2) {
```

Comment:
`canvasCoord1` -> `topLeftCoord`
`canvasCoord2` -> `rightbottomCoord`

Reply:
I think that `insideViewport(topLeftCoord: MapCoord, rightBottomCoord: MapCoord)` is a more intuitive function signature since the caller only need to provide two `MapCoord`. Also, the `mapName` is related to the two map coordinate instead of specified explicitly.

Reply:
`mapName` is already contained in `topleftCoord` and `bottomrightCoord`. If the caller is required to provide `mapName`, he/she may write something as follow:
```javascript
const topleft = new MapCoord('world1', 5, 5);
const bottomright = new MapCoord('world1, 10, 10);
if (insideViewport(topleft.mapName, topleft, bottomright)) {
  ...
}
```

I don't see the point of the `mapName` parameter.

Reply:
`topleftCoord` and `bottomrightCoord` are not MapCoord, while they are just an object containing canvas coordinate. It is not proper to use MapCoord because sometimes a watermark may not be on a point with integer coordinate.

Reply:
It seems that I misunderstood something. The control flow of the current code is "convert mapCoord to canvasCoord" -> "check if canvasCoord lies inside viewport" -> "draw". The control flow in my mind is "check if mapCoord is inside viewport" -> "convert mapCoord to canvas" -> "draw". Therefore, I thought that the parameter type should be `MapCoord`.
I think it is reasonable to use canvas coordinate as the parameter of `insideViewport()`. However, I think the checking of `mapName` should not be contained in `insideViewport()`. A canvas coordinate is not related to any mapName. It is `MapCoord` that is related to a mapName. Imagine that if there is a caller who simply wants to check whether a rectangle (canvas coordinate) lies inside the viewport, he/she would need to manually provide the `mapName` as follows:
```javascript
const topleft = {x: 5, y: 5};
const bottomright = {x: 10, y: -10};
if (insideViewport(this.gameClient.playerInfo.mapCoord.mapName, topleft, bottomright)) {
  ...
}
```
So, my suggestion right now would become removing the `mapCoord` parameter of `insideViewport()` and checking the `mapName` inside the caller.